addons/delivery_shipper/wizard/choose_delivery_carrier.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+
+
+class ChooseDeliveryCarrier(models.TransientModel):
+    _inherit = 'choose.delivery.carrier'
+
+    is_shipper = fields.Boolean(compute='_compute_is_shipper')
+    is_shipper_3pl = fields.Boolean(compute='_compute_is_shipper_3pl')
+    rate_data = fields.Json("Available Rates")
+    rate_line_ids = fields.One2many('shipper.rate.line', 'wizard_id', string="Rates")
+
+    rate_id = fields.Char()
+
+    @api.depends('carrier_id')
+    def _compute_is_shipper(self):
+        self.ensure_one()
+        self.is_shipper = self.carrier_id.delivery_type == "shipper"
+
+    @api.depends('carrier_id')
+    def _compute_is_shipper_3pl(self):
+        self.ensure_one()
+        if self.carrier_id.delivery_type == "shipper":
+            self.is_shipper_3pl = self.carrier_id.shipper_3pl_partner is not False
+        else:
+            self.is_shipper_3pl = False
```

Comment:
```suggestion
        self.is_shipper_3pl = self.carrier_id.delivery_type == 'shipper' and self.carrier_id.shipper_3pl_partner
```
It's easier to read, maybe.
Also, the compute method CANNOT be ensure_one!
You should always write your compute methods expecting that you could receive a record set, even for a wizard.